### PR TITLE
Ports text emphasis from TGMC (who ported it from Citadel)

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Ports text emphasis from TGMC, who ported it from Citadel, who apparently took the idea from Bay/Polaris.
[The apparent original Citadel PR](https://github.com/Citadel-Station-13/Citadel-Station-13/pull/12487)
[The TGMC code I ported directly](https://github.com/tgstation/TerraGov-Marine-Corps/pull/4854)

[](https://i.imgur.com/OBodmPX.png)
[](https://i.imgur.com/ofLdyxD.png)
[](https://i.imgur.com/SMkKcWU.png)

This port differs from TGMC in that you only need to use the characters once on each side to get the effect, rather than twice.

Bold: `+Hi there!+` -> **Hi there!**
Italics: `|What's this?|` -> _What's this?_
Underline: `_I thought I made myself clear..._` -> [well, i can't do underlines here, but you all know what it is]

## Why It's Good For The Game
This gives you more control over how you come across to other people while talking, and lets you emphasize specific parts of sentences without resorting to using !! to bold the entire damn thing and make you yell.

## Changelog

:cl: Kevinz for Citadel, who was ported to TGMC by XSlayer300, now ported to tg by Ryll/Shaps
add: Text modifiers have been added! You can now |italicize| _underline_ and +bold+ your messages.
/:cl:

### CODE REVIEW

**Comment by Couls** [comment]
italicized text looks kind of messy to read

**Comment by Mothblocks** [comment]
thats because italicized text is inaccessible unfortunately

**Comment by AdipemDragon** [comment]
How is mixing and matching handled? Do you:

- drop the current emphasis and start the new combined one afterwards? (aka how Discord handles it)
Ex.:      "Oh, +|HELLO|+ |there!|"

- just set it and forget it until you close it? (like a set of parantheses)
Ex.:      "Oh, +|HELLO+ there!|" 

- arrange them in a way I can't think of right now?

**Comment by Ryll-Ryll** [comment]
> How is mixing and matching handled? Do you:
> 
> * drop the current emphasis and start the new combined one afterwards? (aka how Discord handles it)
>   Ex.:      "Oh, +|HELLO|+ |there!|"
> * just set it and forget it until you close it? (like a set of parantheses)
>   Ex.:      "Oh, +|HELLO+ there!|"
> * arrange them in a way I can't think of right now?

[](https://i.imgur.com/qP0JRKc.png)
`|You can also +mix and _match with| how you use the+ modifiers!_`

File: code/game/say.dm [hunk 1]
```diff
@@ -99,6 +99,18 @@ GLOBAL_LIST_INIT(freqtospan, list(
 	var/spanned = attach_spans(input, spans)
 	return "[say_mod(input, message_mods)], \"[spanned]\""
 
+#define ENCODE_HTML_EMPHASIS(input, char, html, varname) \
+	var/static/regex/##varname = regex("[char](.+?)[char]", "g");\
+	input = varname.Replace_char(input, "<[html]>$1</[html]>")
+
+/atom/movable/proc/say_emphasis(input)
```
Comment (Mothblocks) [inline code/game/say.dm:106]: Documen tit.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Document say_emphasis and then it's good

**Comment by Ryll-Ryll** [comment]
> Document say_emphasis and then it's good

what if i don't know what it does

**Comment by Mothblocks** [comment]
just write my name then

**Review by Mothblocks — APPROVED** [review]
yeabh alirght

**Comment by Ryll-Ryll** [comment]
(to future recruiters combing my PRs: I did know what it does that was just a bit)

**Pull request merged**